src/grainy/core.py

```diff
@@ -2,16 +2,20 @@
 core functionality
 """
 
+from __future__ import annotations
```

Comment:
Not sure this is needed?

Reply:
I needed that to add hints for:
https://github.com/20c/grainy/blob/593603c50b6302cff60440ce2d1ac93f25830cc8/src/grainy/core.py#L95-L102
Couldn't add hint for returning `Namespace` within `Namespace` class without it. Not sure if there is another way.

Reply:
Confirmed that this is currently the way to handle such instances (method returning the enclosing class) from python3.7 all the way up to and including python3.10